projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/JobImageBuilder.java

```diff
@@ -253,14 +256,26 @@ public boolean isBuildingJobInProgress(String dataJobName) {
   private Map<String, String> getBuildParameters(DataJob dataJob, JobDeployment jobDeployment) {
     String jobName = dataJob.getName();
     String jobVersion = jobDeployment.getGitCommitSha();
+    String baseImage = null;
+    String pythonVersion = jobDeployment.getPythonVersion();
+
+    // TODO: Remove this part when datajobs.deployment.dataJobBaseImage is deprecated.
+    if (deploymentDataJobBaseImage != null) {
+      baseImage = deploymentDataJobBaseImage;
+    } else if (pythonVersion != null
+        && supportedPythonVersions.isPythonVersionSupported(pythonVersion)) {
+      baseImage = supportedPythonVersions.getJobBaseImage(pythonVersion);
+    } else {
+      baseImage = supportedPythonVersions.getDefaultJobBaseImage();
+    }
```

Comment:
nit: this logic may be encapsulated, so that the base image is determined with 1 call, via dedicated functionality to fetch the effective base image provided; then you may add a unit test on it

Reply:
Done. Thank you for the review